mteb/models/sentence_transformer_wrapper.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+from typing import Any, get_args
+
+import numpy as np
+import torch
+from sentence_transformers import CrossEncoder, SentenceTransformer
+
+import mteb
+from mteb.abstasks.TaskMetadata import TASK_TYPE
+from mteb.encoder_interface import PromptType
+
+logger = logging.getLogger(__name__)
+
+
+class SentenceTransformerWrapper:
+    def __init__(
+        self,
+        model: str | SentenceTransformer | CrossEncoder,
+        revision: str | None = None,
+        task_to_prompt_name: dict[str, str] | None = None,
+        model_prompts: dict[str, str] | None = None,
+        override_prompts: bool = False,
+        **kwargs,
+    ) -> None:
+        """Wrapper for SentenceTransformer models.
+
+        Args:
+            model: The SentenceTransformer model to use. Can be a string (model name), a SentenceTransformer model, or a CrossEncoder model.
+            revision: The revision of the model to use.
+            task_to_prompt_name: A dictionary mapping task names to prompt names.
+                First priority is given to the composed prompt of task name + prompt type (query or passage), then to the specific task prompt,
+                then to the composed prompt of task type + prompt type, then to the specific task type prompt,
+                and finally to the specific prompt type.
+            model_prompts: A dictionary mapping prompt names with prompts. If not provided, the model's prompts will be used.
+            override_prompts: Whether to override the model's existing prompts with the provided model_prompts.
+            **kwargs: Additional arguments to pass to the SentenceTransformer model.
+        """
+        if isinstance(model, str):
+            self.model = SentenceTransformer(
+                model, revision=revision, trust_remote_code=True, **kwargs
+            )
+        else:
+            self.model = model
+        self.task_to_prompt_name = (
+            validate_task_to_prompt_name(task_to_prompt_name)
+            if task_to_prompt_name
+            else None
+        )
+        if hasattr(self.model, "prompts") and task_to_prompt_name is not None:
+            if model_prompts is None:
+                model_prompts = {v: v + " " for v in task_to_prompt_name.values()}
+            if len(self.model.prompts) == 0:
+                self.model.prompts = model_prompts
+                logger.info(f"Model prompts will be overrided with {model_prompts}")
+            elif override_prompts:
+                # override existing model prompts
+                self.model.prompts = model_prompts
+                logger.info(f"Model prompts will be overrided with {model_prompts}")

```

Comment:
```suggestion
                logger.info(f"Model prompts will be overrided with {model_prompts}")

```suggestion
                logger.info(f"Model prompts will be overwritten with {model_prompts}")

```